dot/core/service.go

```diff
@@ -438,15 +438,26 @@ func (s *Service) maintainTransactionPool(block *types.Block) {
 }
 
 // InsertKey inserts keypair into the account keystore
-// TODO: define which keystores need to be updated and create separate insert funcs for each (#1850)
-func (s *Service) InsertKey(kp crypto.Keypair) {
-	s.keys.Acco.Insert(kp)
+func (s *Service) InsertKey(kp crypto.Keypair, keystoreType string) error {
+	ks, err := s.keys.GetKeystore([]byte(keystoreType))
+	if err != nil {
+		return err
+	}
+
+	ks.Insert(kp)
+
+	return nil
 }
 
 // HasKey returns true if given hex encoded public key string is found in keystore, false otherwise, error if there
-//  are issues decoding string
-func (s *Service) HasKey(pubKeyStr, keyType string) (bool, error) {
-	return keystore.HasKey(pubKeyStr, keyType, s.keys.Acco)
+// are issues decoding string
+func (s *Service) HasKey(pubKeyStr, keystoreType string) (bool, error) {
+	ks, err := s.keys.GetKeystore([]byte(keystoreType))
+	if err != nil {
+		return false, err
+	}
+
+	return keystore.HasKey(pubKeyStr, keystoreType, ks)
```

Comment:
nit maybe instead pass in the keystore as an interface to `InsertKey` and `HasKey` instead of passing its string type?

Also should we use typed strings for the key store type, together with enums (aka iota)?

Reply:
> Also should we use typed strings for the key store type, together with enums (aka iota)?

That is the convention. I generally try to do this when possible. I tried for this PR as well, but that become a rather big change because it impact a lot of things.
Currently, it is not that bad though. There is a `Name` type.